From: Chao Xu <amos.xuchao@gmail.com>
Subject: mm/vmscan: simplify the nr assignment logic for pages to scan
Date: Thu, 10 Nov 2022 19:31:30 +0800

By default the assignment logic of anonymouns or file inactive pages and
active pages to scan using the same duplicated code snippet.  To simplify
the logic, merge the same part.

Link: https://lkml.kernel.org/r/20221110113130.284290-1-Chao.Xu9@zeekrlife.com
Signed-off-by: Chao Xu <Chao.Xu9@zeekrlife.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/vmscan.c |   13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

--- a/mm/vmscan.c~mm-vmscan-simplify-the-nr-assignment-logic-for-pages-to-scan
+++ a/mm/vmscan.c
@@ -5950,14 +5950,11 @@ static void shrink_lruvec(struct lruvec
 		 * scan target and the percentage scanning already complete
 		 */
 		lru = (lru == LRU_FILE) ? LRU_BASE : LRU_FILE;
-		nr_scanned = targets[lru] - nr[lru];
-		nr[lru] = targets[lru] * (100 - percentage) / 100;
-		nr[lru] -= min(nr[lru], nr_scanned);
-
-		lru += LRU_ACTIVE;
-		nr_scanned = targets[lru] - nr[lru];
-		nr[lru] = targets[lru] * (100 - percentage) / 100;
-		nr[lru] -= min(nr[lru], nr_scanned);
+		for ( ; lru <= lru + LRU_ACTIVE; lru += LRU_ACTIVE) {
+			nr_scanned = targets[lru] - nr[lru];
+			nr[lru] = targets[lru] * (100 - percentage) / 100;
+			nr[lru] -= min(nr[lru], nr_scanned);
+		}
 
 		scan_adjusted = true;
 	}
_
